iroh/src/sync_engine/rpc.rs

```diff
@@ -236,17 +218,19 @@ impl<S: Store> SyncEngine<S> {
     ) -> impl Stream<Item = RpcResult<DocGetManyResponse>> {
         let DocGetManyRequest { doc_id, filter } = req;
         let (tx, rx) = flume::bounded(ITER_CHANNEL_CAP);
-        let store = self.store.clone();
-        self.rt.main().spawn_blocking(move || {
-            let ite = store.get_many(doc_id, filter);
-            let ite = inline_result(ite).map_ok(|entry| DocGetManyResponse { entry });
-            for entry in ite {
-                if let Err(_err) = tx.send(entry) {
-                    break;
-                }
+        let sync = self.sync.clone();
+        // we need to spawn a task to send our request to the sync handle, because the method
+        // itself must be sync.
```

Comment:
why must this function be sync?

Reply:
because quic-rpc forces us to return `impl Stream`, not `impl Future`